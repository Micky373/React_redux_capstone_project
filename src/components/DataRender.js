import React from "react";
import { Link } from "react-router-dom";

const DataRender = ({ id, country }) => (
  <>
    <Link to={`/Details/${id}`}>
      <p>{country}</p>
    </Link>
  </>
)

export default DataRender