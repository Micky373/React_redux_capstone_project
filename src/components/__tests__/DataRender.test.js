import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import store from '../../redux/configureStore';
import DataRender from '../DataRender';
import '@testing-library/jest-dom';

describe('DataRender componenet testing', () => {
  const element = (
    <Provider store={store}>
      <HashRouter>
        <DataRender total={'100000'} country="Ethiopia" />
      </HashRouter>
    </Provider>
  );

  test('DataRender snapshot', () => {
    const dataRender = renderer.create(element);
    expect(dataRender).toMatchSnapshot();
  });

  test('Render check', () => {
    render(element);
    const dataRender = screen.getByText('Ethiopia');
    expect(dataRender).toBeInTheDocument();
  });
});
