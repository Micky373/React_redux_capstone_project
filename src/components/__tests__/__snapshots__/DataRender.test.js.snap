// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataRender componenet testing DataRender snapshot 1`] = `
<a
  href="#/Details/2"
  onClick={[Function]}
>
  <p>
    Ethiopia
  </p>
</a>
`;
